### PR TITLE
Bug 1299946 - Add some tests for filters.js

### PR DESCRIPTION
**Pull request by KWierso**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1828)

### CODE REVIEW

**Comment by KWierso** [comment]
I plan on putting a lot more tests in for the highlightCommonTerms section before landing this to hopefully make sure we're highlighting everything we should match and nothing we should not match, but ran out of time today.

**Comment by KWierso** [comment]
Okay, this adds tests for all of the filters except for linkifyClassifications, alertStatus, and displayPrecision.
linkifyClassifications keeps saying repo is null, even though I'm passing in the exact same things that the UI actually does.

I'm not sure what the perfherder filters are supposed to be doing. Maybe @wlach could provide some help there?

File: tests/ui/unit/filtersSpec.js [hunk 1]
```diff
@@ -1,19 +1,245 @@
...
+describe('displayPrecision filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('does some extra things', function() {
+        var displayPrecision = $filter('displayPrecision');
+        expect(displayPrecision('')).toEqual('');
```
Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: The following would be a good start:

```js
expect(displayPrecision('123.53222')).toEqual('123.53');
expect(displayPrecision(1/0)).toEqual('N/A');
```

File: tests/ui/unit/filtersSpec.js [hunk 2]
```diff
@@ -1,19 +1,245 @@
...
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('does some more things', function() {
+        var alertStatus = $filter('alertStatus');
+        expect(alertStatus('')).toEqual('');
+    });
+});
```
Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: I believe the alert status filter isn't actually used, so you can just skip it. :)

**Comment by wlach** [comment]
This looks great! ++

File: tests/ui/unit/filtersSpec.js [hunk 3]
```diff
@@ -1,19 +1,245 @@
...
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('linkifies classifications', function() {
+        var linkifyClassifications = $filter('linkifyClassifications');
+        // This keeps saying "repo is null"...
+        //expect(linkifyClassifications('This 1234567890ab is linked', 'try')).toEqual('');
```
Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: I think $rootScope.repos needs to be defined for this to work:

https://github.com/mozilla/treeherder/blob/master/ui/js/models/repository.js#L18

File: tests/ui/unit/filtersSpec.js (hunk 1 above)
Comment (KWierso) [inline tests/ui/unit/filtersSpec.js]: > expect(displayPrecision(1/0)).toEqual('N/A');
This one actually ends up being 'Infinity', not 'N/A'. Does that sound right to you?

Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: No, but that's a good extra test case. :)

Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: (i.e. leave it in)

File: tests/ui/unit/filtersSpec.js [hunk 4]
```diff
@@ -1,19 +1,243 @@
...
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('does some extra things', function() {
+        var displayPrecision = $filter('displayPrecision');
+        expect(displayPrecision('123.53222')).toEqual('123.53');
+        expect(displayPrecision(1/0)).toEqual('Infinity');
```
Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: Could you add this one as well:

```js
expect(displayPrecision(Number.NaN)).toEqual('N/A');
```

File: tests/ui/unit/filtersSpec.js [hunk 5]
```diff
@@ -1,19 +1,243 @@
...
+});
+
+describe('displayPrecision filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('does some extra things', function() {
```
Comment (wlach) [inline tests/ui/unit/filtersSpec.js]: does some extra things -> returns expected values

File: tests/ui/unit/filtersSpec.js [hunk 6]
```diff
@@ -1,19 +1,244 @@
...
+    });
+
+    it('initializes a two-word name', function() {
+        var initials = $filter('initials');
+        expect(initials('Optimus Prime')).toEqual('<span class="label label-initials">OP</span>');
+    });
+
+    it('initializes a three-word name', function() {
+        var initials = $filter('initials');
+        expect(initials('Some Other Transformer')).toEqual('<span class="label label-initials">ST</span>');
```
Comment (camd) [inline tests/ui/unit/filtersSpec.js]: <peter-griffin-voice>Heh heh heh...  robots in disguise...</peter-griffin-voice>

**Pull request merged**